hive-metastore/src/main/java/org/apache/iceberg/hive/HiveClientPoolProvider.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.hive;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import java.util.concurrent.TimeUnit;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.relocated.com.google.common.annotations.VisibleForTesting;
+
+public class HiveClientPoolProvider {
```

Comment:
Could this object implement the `ClientPool` interface and maybe renamed to `CachedClientPool`?

This way we can push this `ClientPool` to the HiveTableOperations and do not need to change there

Reply:
Maybe even something like this:
https://github.com/apache/iceberg/blob/94705c6799f3b283b45e47e6f9b46b5d081ed666/core/src/main/java/org/apache/iceberg/CachingCatalog.java#L35-L37

Reply:
Refactored it.